reference/test/preconditioner/batch_jacobi_kernels.cpp

```diff
@@ -0,0 +1,237 @@
+        block_ptrs.get_data()[3] = 6;
+
+        b = gko::batch::initialize<BMVec>(
+            {{-2.0, 9.0, 4.0, 1.0, 5.0, 11.0}, {-3.0, 5.0, 3.0, 8.0, 9.0, 7.0}},
+            this->exec);
+        x = BMVec::create(this->exec,
+                          gko::batch_dim<2>(2, gko::dim<2>(this->nrows, 1)));
+    }
+
+    std::shared_ptr<const gko::ReferenceExecutor> exec;
+
+    const size_t nbatch = 2;
+    const int nrows = 6;
+    std::shared_ptr<const Mtx> mtx;
+    const gko::uint32 max_block_size = 3u;
+    gko::array<int> block_ptrs;
+    std::unique_ptr<const BMVec> b;
+    std::unique_ptr<BMVec> x;
+
+    std::unique_ptr<Mtx> get_matrix()
+    {
+        auto mat = Mtx::create(
+            exec, gko::batch_dim<2>(nbatch, gko::dim<2>(nrows, nrows)), 12);
+        int* const row_ptrs = mat->get_row_ptrs();
+        int* const col_idxs = mat->get_col_idxs();
+        value_type* const vals = mat->get_values();
+        // clang-format off
+		row_ptrs[0] = 0; row_ptrs[1] = 2; row_ptrs[2] = 4; row_ptrs[3] = 6; 
+        row_ptrs[4] = 8; row_ptrs[5] = 10; row_ptrs[6] = 12;
+
+		col_idxs[0] = 0; col_idxs[1] = 1;  col_idxs[2] = 0; col_idxs[3] = 1;
+		col_idxs[4] = 0; col_idxs[5] = 2;  col_idxs[6] = 1; col_idxs[7] = 3;
+        col_idxs[8] = 2; col_idxs[9] = 4; col_idxs[10] = 3; col_idxs[11] = 5;
+
+		vals[0] = 2.0; vals[1] = 0.25; vals[2] = -1.0; vals[3] = -3.0;
+		vals[4] = 2.0; vals[5] = 0.2;  vals[6] = -1.5; vals[7] = 0.55; 
+        vals[8] = -1.0; vals[9] = 4.0; vals[10] = 2.0; vals[11] = -0.25;
+        vals[12] = 5.0; vals[13] = 4.25; vals[14] = -7.0; vals[15] = -3.0;
+		vals[16] = 2.0; vals[17] = 0.28; vals[18] = -1.5; vals[19] = 1.55; 
+        vals[20] = -1.0; vals[21] = 4.0; vals[22] = 21.0; vals[23] = -0.95;
+        // clang-format on
+        return mat;
+    }
+};
+
+TYPED_TEST_SUITE(BatchJacobi, gko::test::ValueTypes);
+
+
+TYPED_TEST(BatchJacobi,
+           BatchScalarJacobiApplyToSingleVectorIsEquivalentToUnbatched)
+{
+    using value_type = typename TestFixture::value_type;
+    using Mtx = typename TestFixture::Mtx;
+    using BMVec = typename TestFixture::BMVec;
+    using Jac = typename TestFixture::Jac;
+    using BJac = typename TestFixture::BJac;
+    auto umtxs = gko::test::share(gko::batch::unbatch<Mtx>(this->mtx.get()));
+    auto ub = gko::batch::unbatch<BMVec>(this->b.get());
+    auto ux = gko::batch::unbatch<BMVec>(this->x.get());
+    auto unbatch_prec_fact =
+        Jac::build().with_max_block_size(1u).on(this->exec);
+    for (size_t i = 0; i < umtxs.size(); i++) {
+        auto unbatch_prec = unbatch_prec_fact->generate(umtxs[i]);
+        unbatch_prec->apply(ub[i].get(), ux[i].get());
+    }
+    auto prec_fact = BJac::build().with_max_block_size(1u).on(this->exec);
+    auto prec = prec_fact->generate(this->mtx);
+    value_type* blocks_arr = nullptr;
+    int* block_ptr = nullptr;
+    int* row_part_of_which_block_arr = nullptr;
+    int* cumul_block_storage = nullptr;
```

Comment:
not sure we need those

Reply:
I think it makes the code easier to read than just passing in `nullptr` everywhere.